staging/src/k8s.io/apiserver/pkg/storage/cacher/delegator.go

```diff
@@ -258,3 +290,158 @@ func (c *CacheDelegator) ReadinessCheck() error {
 func (c *CacheDelegator) RequestWatchProgress(ctx context.Context) error {
 	return c.storage.RequestWatchProgress(ctx)
 }
+
+func (c *CacheDelegator) Stop() {
+	c.stopOnce.Do(func() {
+		close(c.stopCh)
+	})
+	c.wg.Wait()
+}
+
+func newConsistencyChecker(resourcePrefix string, newListFunc func() runtime.Object, cacher storeCache, etcd store, clock clock.Clock) *consistencyChecker {
+	return &consistencyChecker{
+		resourcePrefix: resourcePrefix,
+		newListFunc:    newListFunc,
+		cacher:         cacher,
+		etcd:           etcd,
+		clock:          clock,
+	}
+}
+
+type consistencyChecker struct {
+	resourcePrefix string
+	newListFunc    func() runtime.Object
+
+	cacher storeCache
+	etcd   store
+	clock  clock.Clock
+}
+
+type storeCache interface {
+	store
+	Ready() bool
+}
+
+type store interface {
```

Comment:
Why do we need to introduce this interface - having storage.Interface here seems good enough to me.

Reply:
Removed

Reply:
I'm defining a minimal interface needed to be implemented to check consistency, a recommended Go practice. Currently cacher no longer implements the full storage interface. I removed the methods that we just passthrough to store.